src/runtime/GlobalObjectBuiltinObject.cpp

```diff
@@ -327,6 +338,52 @@ static Value builtinObjectFreeze(ExecutionState& state, Value thisValue, size_t
     return O;
 }
 
+static Value builtinObjectFromEntries(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    if (argv[0].isUndefinedOrNull()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().object.string(), true, state.context()->staticStrings().fromEntries.string(), errorMessage_GlobalObject_ThisUndefinedOrNull);
+    }
+    Value iterable = argv[0];
+    Object* obj = new Object(state);
+    FunctionObject* stepsDefine = new NativeFunctionObject(state, NativeFunctionInfo(state.context()->staticStrings().Empty, createDataPropertyOnData, 2, NativeFunctionInfo::Strict));
+
+    Value iterableRecord = getIterator(state, iterable);
+    while (true) {
+        Value next = iteratorStep(state, iterableRecord);
+        if (next.isFalse()) {
+            return obj;
+        }
+        Value nextItem = iteratorValue(state, next);
+        try {
+            // If Type(nextItem) is not Object, then
+            if (!nextItem.isObject()) {
+                // Let error be Completion{[[Type]]: throw, [[Value]]: a newly created TypeError object, [[Target]]: empty}.
+                // Return ? IteratorClose(iter, error).
+                ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "Iterator value a is not an entry object");
+            }
```

Comment:
From the [standard](https://www.ecma-international.org/ecma-262/10.0/#sec-add-entries-from-iterable)
> If Type(nextItem) is not Object, then
    Let error be ThrowCompletion(a newly created TypeError object).
    Return ? IteratorClose(iteratorRecord, error).

IMO I think that instead of throwing the error and catching it with try-catch statement, we have to pass the error into `iteratorClose` method and throw the exception in the `iteratorClose`. We may have to revise the iterator operations. What do you think about it?

Reply:
I like the idea. I can make a follow-up patch on the iteratorClose method.